controller/token.go

```diff
@@ -342,41 +343,104 @@ func (c *TokenController) Exchange(ctx *app.ExchangeTokenContext) error {
 	if payload == nil {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("payload", "nil").Expected("not empty payload"))
 	}
-	if payload.ClientID == nil {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("client_id", "nil").Expected("Service Account ID"))
-	}
-	if payload.ClientSecret == nil {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("client_secret", "nil").Expected("Service Account secret"))
-	}
+	if payload.GrantType == "client_credentials" {
+		if payload.ClientID == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("client_id", "nil").Expected("Service Account ID"))
+		}
+		if payload.ClientSecret == nil {
```

Comment:
we could make this optional for now, for `grant_type=authorization_code`

Reply:
We need it here, for `grant_type=client_credentials`
it is optional for `grant_type=authorization_code`